### PR TITLE
FTA updates to align Trey Research to the other ES reference Implementation

### LINKED ISSUE #526: Ability to filter the region for the first virtual network hub -- Quality of Life
**Issue**
As the Regions increase, so does the scrolling. I understand it is not a top priority, but would like it to function like the Basic tab (example below). The list is just bad, thus allowing the filter option would improve the quality of life. 

image

**Describe the solution you'd like**
This is code is just an example, could have errors, or not work as described. 
```json
{
                        "name": "esNwLocation",
                        "type": "Microsoft.Common.DropDown",
                        "label": "Region for the first virtual network hub",
                        "toolTip": "Select the target region for you connectivity deployment (requires you to provide a subscriptionId for connectivity)",
                        "multiselect": false,
                        "selectAll": false,
                        "filter": true,
                        "filterPlaceholder": "Filter items ...",
                        "multiLine": true,
                        "constraints": {
                            "allowedValues": "[map(steps('esConnectivityGoalState').esLocationsApi.value, (item) => parse(concat('{\"label\":\"', item.displayName, '\",\"value\":\"', item.name, '\"}')))]",
                            "required": true
                        },
                        "visible": "[equals(steps('esConnectivityGoalState').esHub, 'Yes')]"
                    }
```
image

### PR DESCRIPTION
**Pull request by aarunraaj**
**This PR fixes #526 **
ESLZ - Trey Research implementation has been updated with the new policies and changes that are inline with the other RIs updates.

Updated with a new Azure Policy initiative which configures the settings for Azure Security Center.

Security Contacts on subscriptions
continuous export from ASC to central Log Analytics workspace
Azure Policy initiative for TLS/SSL (assigned at landing zones mg)
Azure Policy initiative for denying usage of PaaS services with public endpoints (assigned at corp landing zone)
New enhanced and connectivity setup, where users can now:

Select zone redundant deployment of VPN and ExpressRoute gateways to regions where AZ is supported
Select appropriate SKU for VPN and ExpressRoute - given the conditions above
Option to use Azure Firewall as DNS Proxy. Also, when deploying Azure Firewall, a Firewall Policy will be deployed and will be linked to the Firewall.
Option to deploy DDoS Standard, and to protect landing zones with the DDoS Standard plan

### CODE REVIEW

**Comment by aarunraaj** [comment]
ESLZ - Trey Research implementation has been updated with the new policies and changes that are inline with the other RIs updates.

Updated with a new Azure Policy initiative which configures the settings for Azure Security Center.
1. Security Contacts on subscriptions
2. continuous export from ASC to central Log Analytics workspace
3. Azure Policy initiative for TLS/SSL (assigned at landing zones mg)
4. Azure Policy initiative for denying usage of PaaS services with public endpoints (assigned at corp landing zone)

New enhanced and connectivity setup, where users can now:
 - Select zone redundant deployment of VPN and ExpressRoute gateways to regions where AZ is supported 
- Select appropriate SKU for VPN and ExpressRoute - given the conditions above 
- Option to use Azure Firewall as DNS Proxy. Also, when deploying Azure Firewall, a Firewall Policy will be deployed and will be linked to the Firewall. 
- Option to deploy DDoS Standard, and to protect landing zones with the DDoS Standard plan

File: docs/reference/treyresearch/README.md [hunk 1]
```diff
@@ -41,9 +41,9 @@ By default, all recommendations are enabled. You must explicitly disable them if
   -	A Log Analytics workspace and an Automation account.
   - Azure Sentinel.
   -	A hub virtual network  
-  -	VPN Gateway
-  - ExpressRoute Gateway (optional)
-  -	Azure Firewall
+  -	VPN Gateway (optional - deployment across Availability Zones)
+  - ExpressRoute Gateway (optional - deployment across Availability Zones)
+  -	Azure Firewall (optional - deployment across Availability Zones)
```
Comment (hjscherer) [inline docs/reference/treyresearch/README.md:46]: Can the section about ASC also be updated (free -> _Azure Defender OFF_ and Standard -> _Azure Defender ON_)

**Review by hjscherer — CHANGES_REQUESTED** [review]

File: docs/reference/treyresearch/README.md (hunk 1 above)
Comment (aarunraaj) [reply docs/reference/treyresearch/README.md:46]: @hjscherer , I have updated the ASC section.

File: docs/reference/treyresearch/armTemplates/auxiliary/diagnosticsAndSecurity.json [hunk 2]
```diff
@@ -129,7 +143,7 @@
         "policyDefinitions": {
             "deployAzureActivityLog": "[concat('/providers/Microsoft.Management/managementGroups/', parameters('topLevelManagementGroupPrefix'), '/providers/Microsoft.Authorization/policyDefinitions/Deploy-Diagnostics-ActivityLog')]",
```
Comment (krnese) [inline docs/reference/treyresearch/armTemplates/auxiliary/diagnosticsAndSecurity.json:144]: Can you pls update the policies.json from Wingtip, and change the 'deployAzureActivityLog' to point towards the built-in policy? (This recently moved from custom to built-in, so we are not maintaining it anymore from here)

Comment (aarunraaj) [reply docs/reference/treyresearch/armTemplates/auxiliary/diagnosticsAndSecurity.json]: @krnese , updating these changes and running a local test. I will update you once the test is complete.

**Review by krnese — APPROVED** [review]

**Review by hjscherer — APPROVED** [review]

**Pull request merged**